### DISCUSSION TITLE
[Feature]: Upload a single photo via mobile app

### DISCUSSION THREAD

**Discussion by Morethanevil** [immich-app/immich#1684] [Feature Request]
### Feature detail

It should be possible to upload a single photo via mobile app. Something like the upload button from the web version

### Platform

Mobile App

**Comment by micogg** [comment]
Is there a way to know when in the timeline this "nice to have" is slated to be worked on?

This project is amazing, it's incredibly responsive and the upload speeds are superb.  I'm waiting to use it full time until this feature is implemented.  I'm using Immich primarily to store my kids pictures, but they are scattered in several different albums on my phone and I don't want to backup all albums so it would be nice to be able to upload specific pictures.

Regardless, great job with the project.

**Comment by alextran1502** [comment]
@micogg We don't have a timeline set for the `nice to have` feature yet.

For your specific needs, you can create an album, let's say "Kid's photos". Then from your phone's gallery, you can select and move all of the photos/videos of your kids that you want to backup to that album/folder.

From Immich, you can select that album to be the only one that will be back up from the album selection screen, then Immich will only upload the photos and videos in that album. If you use Android and enable background backup, every time you move the photos/videos to that album, they will be uploaded in the background.

**Reply by Vcele** [reply]
I think syncing albums makes a lot of sense, but explaining my friends how to create albums to upload photos into our shared photo cloud seems quite complicated. Especially on the Google photos app it's rather complex to create those because the standard way creates a Google photos album and not a local folder. I think the ability to upload selected photos directly in the backup section would be a great feature to simplify the upload process and prevent clutter on the phones when uploading only a few photos  👍

**Comment by micogg** [comment]
Thanks for the reply Alex, I will do that for now and eagerly await the nice to have feature!

**Comment by irmantask** [comment]
Thanks Alex for your amazing work!

> @micogg We don't have a timeline set for the `nice to have` feature yet.
> 
> For your specific needs, you can create an album, let's say "Kid's photos". Then from your phone's gallery, you can select and move all of the photos/videos of your kids that you want to backup to that album/folder.
> 
> From Immich, you can select that album to be the only one that will be back up from the album selection screen, then Immich will only upload the photos and videos in that album. If you use Android and enable background backup, every time you move the photos/videos to that album, they will be uploaded in the background.


Waiting for this feature too.
> Thanks for the reply Alex, I will do that for now and eagerly await the nice to have feature!

**Comment by Vyerni** [comment]
Definitely waiting on this feature. Would be handy for my use case.

**Comment by enboig** [comment]
As I stated at #2440 being able to use Android sharing would be great. It would also allow saving files from social and messaging apps.

**Comment by donjakobo** [comment]
+1 on this one for me too, I would just like the ability to create custom Albums and select by date (all day) or single photo selection what to upload, as I don't need to backup all my random photos, but I do value being able to create shared albums especially across multiple phones without totally backing everything up THEN selecting out of those backups.

**Comment by irmantask** [comment]
I am still waiting this feature to use Immich daily :)

**Comment by BaldFabi** [comment]
Today I actually thought I migrate from photoprism to immich. But I never imagined that one can't select a single image from the phone.
Is this really a nice to have feature?

**Reply by alextran1502** [reply]
Yes it is. You can already select an album or albums for backup, with an inclusive and exclusive mechanism to help you backup anything you need. Individually selecting photos for backing up thus just a UX improvement on top of what we are already have :) Give it a try to see if it fits your needs.

**Reply by irmantask** [reply]
Yeah but this small improvement would make huge impact for many folks (Google Photos app also has this functionality. So, probably from UX persective is not only to nice to have feature)  :)

**Reply by BaldFabi** [reply]
My goal is to use immich as my goto software to share photos with at least my partner. Backing up my whole library on my phone is already done by nextcloud.
If I create an album on my phone just to simple upload all this photos to immich and then I have to additional select them in the immich app seems a little bit unnecessary.
It would really be great if we could just select shortly an image that should be added to an album. This would imho safe a much of time and additional effort.

**Reply by donjakobo** [reply]
I also want to use Immich to just share selected photos, I really used to love the way the old Amazon photos app let you select (on android) entire days worth of photos, that I could easily share with my partner or family when we traveled or for family events where I already am backing up my phone to my NAS and only need to make sharing a priority.

**Reply by alextran1502** [reply]
@irmantask  Nice-to-have, as it is not a high-priority feature we are aiming to tackle at the moment because of the nature of the open-source project. There are lots of things to work on with a handful of contributors, so we have to pick and choose. Of course, the contribution is very welcomed. 

@BaldFabi Hello, have you tried out the mobile application and the upload mechanism to see if the current mechanism cannot provide what you are seeking?

@donjakobo You can already select individual photos for sharing, adding them to albums, favorites, or archives. I think this thread causes some misunderstanding 😛 for the reader. The discussion is for viewing a photo and then clicking an upload button to backup only that specific photo.

**Reply by donjakobo** [reply]
Right, I only wish to select key photos from my mobile phone and add them to an album, without backing everything up, and then selecting from a backed up list of photos.

Basically if I wasn't clear, I want to upload one photo at a time into an album, currently there is no option to do so, and I am forced to only select existing uploaded photos.

**Reply by BaldFabi** [reply]
@alextran1502 Yes I have. I would love to have the same functionality as in the browser to just select one (specific) photo (as @donjakobo described)

And please don't get me wrong. I'm totally thankful for your work, the application etc. but I didn't expect that a function which seems so obvious isn't implemented, at least not in the app where I would definitely need/expect it

**Reply by bo0tzz** [reply]
Immich is very young, and as such a lot of 'basic' stuff hasn't been implemented yet.

**Reply by SitramSoft** [reply]
As a workaround, you can open on your phone browser the immich web page and upload from there any pictures you want from your phone. The only issue I found with this approach is that each app stores photos in a separate folder and it is very difficult to identify where that location is on the phone filesystem.

**Reply by LaVINCENTgit** [reply]
@SitramSoft 

> As a workaround, you can open on your phone browser the immich web page and upload from there any pictures you want from your phone. The only issue I found with this approach is that each app stores photos in a separate folder and it is very difficult to identify where that location is on the phone filesystem.

This is the way i use too... But (donno if only on xiaomi) also if you select multiple photos and videos, only one is uploaded. Do you have the same issue?

**Comment by shenlong-tanwen** [comment]
An initial implementation for this feature is now released in the v1.72.0 release. Feel free to suggest any improvements or issues with the current implementation.
- #3574 

You can open a local only asset and use the upload button to upload it to the server or you can select _up to **30** local assets_ from the timeline and use the upload button from the bottom sheet. The bottom sheet is made dynamic and so, the upload button will be displayed only when all the assets that are selected are local only.

⚠️ The current implementation lacks a cancel button to stop an ongoing upload. I'll try to add it as an action button in the notification and in the backup page in the next release

**Reply by Vcele** [reply]
I have a question about the general functioning of Immich. Wasn't it always the case that only the albums to be uploaded were displayed in the timeline? I am very happy that there is now a single upload feature, but wouldn't it make sense to continue to show only the uploaded albums in the timeline and to allow single uploads via the album page, where non-synchronised albums could also be displayed? In addition, one could then also add the function to display select non-synchronised albums, if someone needs this function.

**Reply by Vcele** [reply]
Actually now it doesn't show all my pictures in the timeline again. Maybe that was just a bug, but I still think that all Albums should be displayed under "Albums on Device", synchronized or not.

**Reply by enboig** [reply]
I was expecting integration in "phone sharing system", so you can send images/videos from any application (messaging, forum, another gallery, ...) to immich 😅

**Reply by shenlong-tanwen** [reply]
> Wasn't it always the case that only the albums to be uploaded were displayed in the timeline?

Yes. However, if a user has not enabled auto backup or have not initiated a backup yet, the assets from the selected albums will be displayed in the timeline as local only assets. In which case, the user can select the local assets and upload them to Immich. This is useful for those who don't want to enable auto backup and don't want to bulk upload assets as well.

If you select 'Recents' in the albums to upload, all your assets, local and remote will be displayed in the timeline and all the albums will be displayed under 'Albums on Device'

**Reply by Gorkycreator** [reply]
>This is useful for those who don't want to enable auto backup and don't want to bulk upload assets as well.
>
> If you select 'Recents' in the albums to upload, all your assets, local and remote will be displayed in the timeline and all the albums will be displayed under 'Albums on Device'

Thanks, this was what I was missing from the intended workflow! I was questioning whether this method was meant to be used with auto-backup permanently disabled. Turning off upload, adding 'Recents', uploading images, removing 'Recents', and enabling upload is clunky, but seems to be valid if users want to juggle auto backup and individual uploads.

When I heard that the feature was implemented, I was also expecting the method @enboig described. The current method is very appreciated, but it's a bit of a configuration pretzel for users to set up. I don't feel comfortable passing it to family members yet, since there's a big "Start Backup" button that'll be primed with potential to accidentally generate situations with lots of manual sorting and removal of items they didn't intend to upload.

**Comment by waclaw66** [comment]
When you add a local asset to an album, you could start upload automatically.

**Comment by nep2ner** [comment]
This is an awesome feature, but I can't seem to find it. Immich doesn't show up in the OS share menu, and I can't find anywhere inside the Immich app to select photos for upload. I can only see photos that are in folders selected for auto-backup, and therefore are already backed up. My app version is 1.72.2. I see in the release notes how it is supposed to look, but I can't seem to find where I can view non-uploaded images  What am I missing here?

**Reply by Gorkycreator** [reply]
Currently, you can only see the button to upload images if you have backup settings disabled. Images in backup folders that have not been uploaded will then show up in the immich app, but have a local-only icon. These can be selected and uploaded manually. More details available in [this thread](https://github.com/immich-app/immich/discussions/1684#discussioncomment-6682654).

**Reply by shenlong-tanwen** [reply]
The following discussion is also about improving this flow as well:
- https://github.com/immich-app/immich/discussions/3826#discussioncomment-6808576

I'm not sure what's the best way to handle this either, but I do know that this has to be improved to make it less clunky.

**Reply by Monarsoft** [reply]
For local deployment, when using the immich mobile app, some videos failed to upload and were repeatedly uploaded without success, resulting in the content behind the iteration not being processed.

**Comment by alex-sherwin** [comment]
Are you open to a PR to add an upload single/multiple file selection from iOS directly into an Immich album, e.g. the standard iOS sharing paradigm most people in this thread were hoping for?

**Comment by architomsn** [comment]
+1 for the possibility to upload a single or multiple assets in the mobile app - like in the web-ui.

![Screenshot_20240924-195249](https://github.com/user-attachments/assets/7bac2120-0047-499f-997b-0d114a1c21bd)

**Reply by shivanshs9** [reply]
@architomsn It's already present, even though it's very confusing on how to use it. Mentioned here - https://github.com/immich-app/immich/discussions/3826#discussioncomment-6808576.

I've just raised a PR as well to highlight this feature in the docs - https://github.com/immich-app/immich/pull/14225 🚀

**Reply by rijndael86** [reply]
This doesn't include a "share to Immich" option when clicking share in the classic Android menu right?

Because if you have files under a hidden folder (starting with a dot), Immich will not see those assets and you won't be able to upload them from mobile. It'd be nice to be able to manually click share when visualizing a picture in other gallery or files app and let the user upload it to the server.

**Comment by ramphex** [comment]
Also waiting for this feature. Getting resistance from friends and family on creating separate albums on their phones for pictures they want to upload since they don’t want to dump their entire photo gallery into my server.

Edit: this post is in relation to the iOS mobile app

**Comment by garypaduana** [comment]
I can't believe "upload single photo or selection of photos" isn't part of the MVP for the Android app. The app should have never been published without it. It's the first thing I looked for upon opening the app for the first time and it's sadly not present. That makes it a 1-star app as it's completely useless and I must use the browser instead.

**Reply by alextran1502** [reply]
@garypaduana You can select individual asset or a group of assets for upload after selecting an album for as target for backup.   

On a side note, I was trying to understand the tone in your comment but I couldn't find a reasonable explanation. So maybe understanding where the app comes from, what it is trying to achieve, and what capabilities it is giving you would give you a better of understanding why some features are missing.

Cheers

**Reply by garypaduana** [reply]
> On a side note, I was trying to understand the tone in your comment but I couldn't find a reasonable explanation. 

It's a CRUD app without create/update functionality.

> We don't have a timeline set for the `nice to have` feature yet.

It's a CRUD app without create/update functionality. It's incomplete. I can't come up with a reasonable explanation for _your_ tone. It's a CRUD app. And we're now years in on this project and you still can't upload a single file. Feedback is a gift, sorry it's not wrapped in the expected praise and thanks.

**Reply by alextran1502** [reply]
@garypaduana 

> It's a CRUD app without create/update functionality.

Thank you for the feedback. I didn't expect the feedback to be wrapped in the "expected" praise and thanks; I am not entitled. The sentence that troubled me the most is, "That makes it a 1-star app as it's completely useless", which is untrue and exaggerated. I hope we create a good environment for giving and receiving feedback so that all the contributors that put their effort into the project, when reading the input, don't have to feel, "shit, man, why such harsh words."

Anyway,

![image](https://github.com/user-attachments/assets/10aecf4a-81a1-4988-9c0f-5a224883f579)


As stated, you can upload a single photo; there is a way to do that. The app is built for mass uploading, so we choose not to have a big upload button on the timeline. The viewing experience is vital to the users and me, so we make some sacrifices on the real estate of the home page. 

I plan to add this upload button somewhere, but I haven't figured out where would be the least intrusive place yet.

**Comment by ericarthurc** [comment]
From my Samsung Galaxy app.. super easy

![Screenshot_20250329_162217_IntentResolver](https://github.com/user-attachments/assets/65c46fdb-f9ac-40db-b0d8-34867678d4a9)

**Comment by HeytalePazguato** [comment]
3 years later, still waiting for this feature

**Reply by Vyerni** [reply]
Can't you just open the app, and select upload in the top right? 

I can

**Reply by AverageHelper** [reply]
@Vyerni isn't that just the Album Backup option? I don't see an option in-app for single or selected image upload..

**Comment by melyux** [comment]
Can we reopen this?

**Reply by bo0tzz** [reply]
No, because it's already implemented, as also discussed upthread.

**Reply by melyux** [reply]
Where? I don't think so.

**Reply by garypaduana** [reply]
I also didn't find it and said the same but then deleted that comment after further investigation. The workaround for uploading a single image is to 

1. go into the `Backup` section (icon next to your user account icon), 
2. select albums from your device that contain pictures you wish to upload, 
3. turn off `Sync albums`, 
4. go back to the main screen `Photos` tab, you should see the images from your selected device album with a slashed cloud icon (not uploaded yet), click on that photo, then select the cloud icon with an up arrow to upload it to the server, 
5. create an album in immich and `Add photos` selecting from the images you previously uploaded.

I think the complaint is people just want to go to an album and have a file picker from the device skipping all of these steps. The "upload a single photo" feature is hidden behind an album synchronization feature set.

**Reply by architomsn** [reply]
> The workaround for uploading a single image is to

No, a workaround isn't necessary:

- select the images you want upload
- "share"
- "upload to immich"

![Screenshot_20250512-233533](https://github.com/user-attachments/assets/a36551da-b734-4425-9b62-391eed423193)

**Reply by melyux** [reply]
Yeah, it's very convoluted. People think selecting an album to sync is going to reveal their entire photo library to the server (understandably).

The share sheet method converts HEIC photos into JPEG on iOS, and also removes the Live Photo video from it. Which is bad

**Reply by bo0tzz** [reply]
> The share sheet method converts HEIC photos into JPEG on iOS, and also removes the Live Photo video from it. Which is bad

I don't use iOS but AFAIK there are options for this that you can select while sharing.

**Reply by melyux** [reply]
You can change the compatibility to "Current" in the share sheet options, but it still strips away the Live Photo video. Can't replace a real upload button in the app as it is

**Reply by shenlong-tanwen** [reply]
> Can't replace a real upload button in the app as it is

We are currently reworking our sync implementation. After the rework, all the albums on the device will be listed under the Library tab group - "On this device". You can then select and upload individual assets from within the local album, even if the album is not selected for backup

**Reply by icf20** [reply]
> > The workaround for uploading a single image is to
> 
> No, a workaround isn't necessary:
> 
>     * select the images you want upload
> 
>     * "share"
> 
>     * "upload to immich"
> 
> 
> ![Screenshot_20250512-233533](https://private-user-images.githubusercontent.com/168718687/442959801-a36551da-b734-4425-9b62-391eed423193.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk0OTE0NjEsIm5iZiI6MTc0OTQ5MTE2MSwicGF0aCI6Ii8xNjg3MTg2ODcvNDQyOTU5ODAxLWEzNjU1MWRhLWI3MzQtNDQyNS05YjYyLTM5MWVlZDQyMzE5My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwOVQxNzQ2MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYWI3NGNhMDNkNzlkYjVmYzc0YjExNzY1NzM1MWI2OTRkMDYzZjU3NjYyOWRlNWEzN2ZlOWIyMjU0M2M4YWE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LU_OzU_VRsGBky1siBi__c42og9Z-ucTNiLoea5n0kI)

this currently fails to upload on current date latest container and android 15

**Comment by architomsn** [comment]
You are right. Upload fails.
Version 1.134.0, build 2200

**Comment by X-lem** [comment]
Just started using immich and the IOS app. Honestly really surprised that this wasn't a feature. Found this thread only to find out it is a feature, but it's the most convoluted method ever (https://github.com/immich-app/immich/discussions/1684#discussioncomment-13122812). No wonder people don't realize it exists (which I argue it still doesn't in the way people actually want).

The way this should be done is there should be a **+** button at the top of the screen next to "backup album" button. Click it, select your photos and album, click upload. That's it. Alex mentioned above:
> The app is built for mass uploading, so we choose not to have a big upload button on the timeline" 

Is anyone asking for a "bug button on the timeline". Reading through this thread and it just seems like people want a simple + button at the top of the banner. That doesn't take up any more room and it's 1000% more obvious than the actual steps. As other's have said it was one of the first things I looked for when trying to upload photos via my phone.

**Reply by shenlong-tanwen** [reply]
A lot of things have changed since the linked reply. A more straightforward solution exists on the app now. You can go to the Library tab -> Albums on this device and can choose your photos and upload them from there

**Reply by convicte** [reply]
Perhaps it's something I am doing wrong, but having the latest Immich app on my Android device following your description I am presented only with a selection of pictures already in Immich repository, to be added to the Album and NOT what this feature was requesting, i.e. selection of images from the device one can upload.

What am I doing wrong that would make your solution work for this?
If not too much to ask, some screenshots of the process, would go a very long way.

**Reply by X-lem** [reply]
I seem to be getting the same thing convicte said. I can only add existing photos in immich to an album. Where can I upload photos from my phone to immich?

**Reply by shenlong-tanwen** [reply]
You can find the "On this device" group in the Library tab from which you can access all your local assets. And if you select the ones that hasn't been uploaded yet, You get the "Upload" action to manually upload them.
| Albums on device | Upload Action |
| :-: | :-: | 
<img width="128" height="285" alt="Albums_On_Device" src="https://github.com/user-attachments/assets/c0832ca6-ecd7-4a47-9a46-f993cac53959" /> | <img width="128" height="285" alt="Upload_Button" src="https://github.com/user-attachments/assets/78cb1723-7983-4d4c-bacf-7dd139644143" />

**Reply by architomsn** [reply]
@shenlong-tanwen 
You are right! That way goes.

**Reply by X-lem** [reply]
Ahhh, I see. Thank you!
